### ISSUE TITLE
User list contains only seen nicks

### ISSUE THREAD

**Issue by rootzlevel** [the-kenny/weechat.el#33]
Users get added or deleted from the list when they join, speak, quit. That's ok. But shouldn't the list be populated with all users of the channel initially?

**Comment by the-kenny** [comment]
Yes, that would be perfect. The reason we implemented this unfortunate behavior is that the parser for the binary data is *very* slow. Weechat sends the nicklist as one very big message which would cause emacs to hang for multiple seconds when joining a *big* channel like #emacs.

I really need to take some time to optimize the parser.

**Comment by jgoerzen** [comment]
I would be OK with a configurable option for this.  I tend to have long running (gnuserv) emacs processes and could live with this.

**Comment by jgoerzen** [comment]
Incidentally, is there a way to simulate the weechat nickbar?  That is, have something visible with a list of users corresponding to the currently-selected frame?  speedbar is halfway this, but it doesn't automatically switch with different buffers.

**Comment by the-kenny** [comment]
I think most of the code is there, we just need to expose it as a configuration option. I'll try to have a look at this soon.

As for the nickbar: I think speedbar (or something similar) is the correct way to handle this. We could handle buffer switches etc. via hooks and get rid of the current shortcomings.
